vendors/circom/circomlib/generated/common/calcwit.cpp

```diff
@@ -213,6 +216,109 @@ Circom_Circuit* loadCircuit(std::string const &datFileName) {
   return circuit;
 }
 
+bool check_valid_number(std::string & s, uint base){
+  bool is_valid = true;
+  if (base == 16){
+    for (uint i = 0; i < s.size(); i++){
+      is_valid &= (
+        ('0' <= s[i] && s[i] <= '9') ||
+        ('a' <= s[i] && s[i] <= 'f') ||
+        ('A' <= s[i] && s[i] <= 'F')
+      );
+    }
+  } else{
+    for (uint i = 0; i < s.size(); i++){
+      is_valid &= ('0' <= s[i] && s[i] < char(int('0') + base));
+    }
+  }
+  return is_valid;
+}
+
+void json2FrElements (json val, std::vector<FrElement> & vval){
+  if (!val.is_array()) {
+    FrElement v;
+    std::string s_aux, s;
+    uint base;
+    if (val.is_string()) {
+      s_aux = val.get<std::string>();
+      std::string possible_prefix = s_aux.substr(0, 2);
+      if (possible_prefix == "0b" || possible_prefix == "0B"){
+        s = s_aux.substr(2, s_aux.size() - 2);
+        base = 2;
+      } else if (possible_prefix == "0o" || possible_prefix == "0O"){
+        s = s_aux.substr(2, s_aux.size() - 2);
+        base = 8;
+      } else if (possible_prefix == "0x" || possible_prefix == "0X"){
+        s = s_aux.substr(2, s_aux.size() - 2);
+        base = 16;
+      } else{
+        s = s_aux;
+        base = 10;
+      }
+      if (!check_valid_number(s, base)){
+        std::ostringstream errStrStream;
+        errStrStream << "Invalid number in JSON input: " << s_aux << "\n";
+	      throw std::runtime_error(errStrStream.str() );
+      }
+    } else if (val.is_number()) {
+        double vd = val.get<double>();
+        std::stringstream stream;
+        stream << std::fixed << std::setprecision(0) << vd;
+        s = stream.str();
+        base = 10;
+    } else {
+        std::ostringstream errStrStream;
+        errStrStream << "Invalid JSON type\n";
+	      throw std::runtime_error(errStrStream.str() );
+    }
+    Fr_str2element (&v, s.c_str(), base);
+    vval.push_back(v);
+  } else {
+    for (uint i = 0; i < val.size(); i++) {
+      json2FrElements (val[i], vval);
+    }
+  }
+}
+
+void loadJson(Circom_CalcWit *ctx, std::string filename) {
+  std::ifstream inStream(filename);
+  json j;
+  inStream >> j;
+
+  u64 nItems = j.size();
+  // printf("Items : %llu\n",nItems);
```

Comment:
Please remove this line. Maybe not since it's code from https://github.com/iden3/circom/blob/f0deda416abe91e5dd906c55507c737cd9986ab5/code_producers/src/c_elements/common/main.cpp#L169?

Reply:
For better maintenance, I'll aim to leave it as unmodified as possible.